### ISSUE TITLE
Add pool max fails logic

### ISSUE THREAD

**Issue by KaymeKaydex** [tarantool/go-tarantool#422]
it is necessary to give and write the logic of behavior in case of unavailability of one of the nodes, I suggest using a similar approach with nginx and the max_fails setting in upstream

https://nginx.org/en/docs/stream/ngx_stream_upstream_module.html#upstream

**Comment by oleg-jukovec** [comment]
I don't think we can give such a guarantee due to an asynchronous nature of the connector: you could push N requests to a connection and only than get an error.

**Comment by KaymeKaydex** [comment]
Yes, we discussed this idea and decided that it is enough on client side
